### PR TITLE
2024 parameter changes

### PR DESCRIPTION
**Pull request by Eric-Sommer**
### What problem do you want to solve?

Implement changes effective 2024 which have not been implemented yet. These [affect](https://www.bundesregierung.de/breg-de/service/gesetzesvorhaben/neuregelungen-januar-2024-2250794)
- Mindestlohn
- Beitragsbemessungsgrenzen
- Bürgergeld
- Gesetzliche Rente

### Todo

- [x] Pick an appropriate title.
- [x] Document PR in CHANGES.md.

### CODE REVIEW

File: src/_gettsim/parameters/ges_rente.yaml [hunk 1]
```diff
@@ -258,13 +258,17 @@ beitragspflichtiges_durchschnittsentgelt:
...
   2022-01-01:
-    scalar: 38901
-    reference: §1 V. v. 30.11.2021, BGBl. I S. 5044.
-    note: Das vorläufige Durchschnittsentgelt.
+    scalar: 42053
+    reference: §3 V. v. 24.11.2023 BGBl. 2023 I Nr. 322
+  2023-01-01:
+    scalar: 43142
+    note: Vorläufiges Durchschnittsentgelt
+  2024:
```
Comment (MImmesberger) [inline src/_gettsim/parameters/ges_rente.yaml:269]:
```suggestion
  2024-01-01:
```

**Review by MImmesberger — APPROVED** [review]
Thanks for the update!

**Review by hmgaudecker — CHANGES_REQUESTED** [review]
Thanks! 

#681 needs to be fixed here, too. And possibly more, are we sure this is the exhaustive list?

**Comment by MImmesberger** [comment]
I am currently writing a test for #681. 
But I didn't check for exhaustiveness of the changes.

File: src/_gettsim/parameters/sozialv_beitr.yaml [hunk 2]
```diff
@@ -881,6 +889,12 @@ mindestlohn:
     scalar: 10.45
   2022-10-01:
     scalar: 12
+  2024-01-01:
+    scalar: 12.41
+    reference: https://www.bundesregierung.de/breg-de/themen/arbeit-und-soziales/mindestlohn-steigt-2223632
+  2025-01-01:
+    scalar: 12.82
+    reference: https://www.bundesregierung.de/breg-de/themen/arbeit-und-soziales/mindestlohn-steigt-2223632
```
Comment (mjbloemer) [inline src/_gettsim/parameters/sozialv_beitr.yaml:897]: source is [Vierte Verordnung zur Anpassung der Höhe des Mindestlohns (Vierte Mindestlohnanpassungsverordnung – MiLoV4) Vom 24. November 2023](https://www.recht.bund.de/bgbl/1/2023/321/VO.html)

```suggestion
  2024-01-01:
    scalar: 12.41
    reference: V. v. 24.11.2023 BGBl. 2023 I Nr. 321
  2025-01-01:
    scalar: 12.82
    reference: V. v. 24.11.2023 BGBl. 2023 I Nr. 321
```

File: src/_gettsim/parameters/sozialv_beitr.yaml [hunk 3]
```diff
@@ -552,6 +552,14 @@ beitr_bemess_grenze_m:
...
+    ges_krankv:
+      west: 5175
+      ost: 5175
+    ges_rentenv:
+      west: 7550
+      ost: 7450
+    reference: V. v. 24.11.2023 BGBl. 2023 I Nr. 322
 bezugsgröße_selbst_m:
   name:
     de: Monatliche Bezugsgröße
```
Comment (mjbloemer) [inline src/_gettsim/parameters/sozialv_beitr.yaml:565]: This has been updated for 2024 as well: https://www.recht.bund.de/bgbl/1/2023/322/VO.html

**Review by mjbloemer — COMMENTED** [review]
Current changes in this MR are correct. Added one source.

Some missing things for 2024 so far:
- Grundfreibetrag 11784 Euro
- Kinderfreibetrag 9540 Euro
- Höchstbetrag Kinderzuschlag 292 Euro
- Minijob 538 Euro

**Comment by MImmesberger** [comment]
I added a test case which reproduces #681. Not sure about the parameters needed to hit the max transfer, though.

**Comment by mjbloemer** [comment]
> Grundfreibetrag and Kinderfreibetrag was formulated in 2022 already and is implemented in gettsim.

Yes, but this is outdated as Existenzminimum (which is the reason for the changes in Regelsätze, Kinderzuschlag, Mindestunterhalt,...) is now higher than expected. The resulting change for Grundfreibetrag and Kinderfreibetrag was discussed in September (see, e.g. https://www.faz.net/aktuell/wirtschaft/grundfreibetrag-soll-auf-11-784-euro-steigen-19161886.html) but I don't think there is a law yet.

File: src/_gettsim/parameters/sozialv_beitr.yaml (hunk 2 above)
Comment (Eric-Sommer) [reply src/_gettsim/parameters/sozialv_beitr.yaml:897]: Thanks, didn't find that one

**Comment by Eric-Sommer** [comment]
> > Grundfreibetrag and Kinderfreibetrag was formulated in 2022 already and is implemented in gettsim.
> 
> Yes, but this is outdated as Existenzminimum (which is the reason for the changes in Regelsätze, Kinderzuschlag, Mindestunterhalt,...) is now higher than expected. The resulting change for Grundfreibetrag and Kinderfreibetrag was discussed in September (see, e.g. https://www.faz.net/aktuell/wirtschaft/grundfreibetrag-soll-auf-11-784-euro-steigen-19161886.html) but I don't think there is a law yet.

This might be the reason for #681 . If there is now law yet, we don't have much parameters to model.

**Comment by mjbloemer** [comment]
> > > Grundfreibetrag and Kinderfreibetrag was formulated in 2022 already and is implemented in gettsim.
> > 
> > 
> > Yes, but this is outdated as Existenzminimum (which is the reason for the changes in Regelsätze, Kinderzuschlag, Mindestunterhalt,...) is now higher than expected. The resulting change for Grundfreibetrag and Kinderfreibetrag was discussed in September (see, e.g. https://www.faz.net/aktuell/wirtschaft/grundfreibetrag-soll-auf-11-784-euro-steigen-19161886.html) but I don't think there is a law yet.
> 
> This might be the reason for #681 . If there is now law yet, we don't have much parameters to model.

Sure. No need to add it now.

(However, another source for the expected new value for Kinderfreibetrag can already be found here: https://www.bmfsfj.de/bmfsfj/themen/familie/familienleistungen/freibetraege-fuer-kinder/freibetraege-fuer-kinder-73890)

**Comment by ChristianZimpelmann** [comment]
> Grundfreibetrag 11784 Euro

@mjbloemer, you mean for Einkommensteuer? According to [this](https://www.bundesfinanzministerium.de/Content/DE/Standardartikel/Themen/Steuern/das-aendert-sich-2024.html#:~:text=Der%20Grundfreibetrag%20wird%20erh%C3%B6ht%3A%20F%C3%BCr,auch%20zu%20einer%20geringeren%20Lohnsteuer.), it is 11604, as specified at the moment.

Or did you mean "steuerfreies Existenzminimum"?

**Comment by MImmesberger** [comment]
@ChristianZimpelmann and I haven't found the parameters to base the maximum Kinderzuschlag calculation on. Hence, we hard-coded the maximum KiZ to be 292€ (as it was done in 2023 and before 2021). @mjbloemer do you happen to know whether that is correct? Or is the calculation based on parameters we don't have access to?

**Comment by mjbloemer** [comment]
> @ChristianZimpelmann and I haven't found the parameters to base the maximum Kinderzuschlag calculation on. Hence, we hard-coded the maximum KiZ to be 292€ (as it was done in 2023 and before 2021). @mjbloemer do you happen to know whether that is correct? Or is the calculation based on parameters we don't have access to?

If you take the new Kinderfreibetrag I stated above, one could approximately replicate this as:

```
(2* Kinderfreibetrag für das sächliche Exmin - BuT)/12 - Kindergeld = (2*3306-336)/12-250 = 273
```

plus 20 Euro Kindersofortzuschlag.

But the calculation it does not matter if the value 292 is simply stated in a law.

**Comment by mjbloemer** [comment]
> > Grundfreibetrag 11784 Euro
> 
> @mjbloemer, you mean for Einkommensteuer? According to [this](https://www.bundesfinanzministerium.de/Content/DE/Standardartikel/Themen/Steuern/das-aendert-sich-2024.html#:~:text=Der%20Grundfreibetrag%20wird%20erh%C3%B6ht%3A%20F%C3%BCr,auch%20zu%20einer%20geringeren%20Lohnsteuer.), it is 11604, as specified at the moment.
> 
> Or did you mean "steuerfreies Existenzminimum"?
> 
> See also [here](https://amp2.wiwo.de/finanzen/steuern-recht/steuererklaerung-diese-freibetraege-pauschalen-und-freigrenzen-gelten-2024/27825826.html):
> 
> > Für 2024 ist zunächst ein Grundfreibetrag von 11604 Euro vorgesehen. Allerdings könnte sich letzterer im Laufe des Jahres noch einmal ändern. So gab es zuletzt Überlegungen den Grundfreibetrag angesichts der hohen Inflation noch einmal anzupassen, um Verbraucher zu entlasten.  Angedacht ist, dass der Grundfreibetrag für 2024 rückwirkend auf 11784 Euro festgelegt werden soll. Beschlossen ist das aber noch nicht. Das Vorhaben wird voraussichtlich erst im Frühjahr umgesetzt und soll dann nachträglich zum 1. Januar 2024 wirken.

I mean Grundfreibetrag and Kinderfreibetrag. Another media source [here](https://www.rnd.de/politik/neuer-ampel-streit-um-kindergeld-worum-geht-es-H5YM26HSB5HIDP6UQGB3VLLP3U.html). Both changes are simply given by the new Existenzminimum.

I agree with Eric that we do not *have to* add this to the main branch yet as there is no law yet.

**Comment by MImmesberger** [comment]
To summarize, so far we implemented:

- [ ] Increase in Mindestlohn
- [ ] Increases in the Bürgergeld Regelsatz
- [ ] Increase in Kinderfreibetrag (not yet in law, but announced by BMF)
- [ ] Vorläufiges Durschnittsentgelt of gesetzliche Rente
- [ ] (hard-coded) Kinderzuschlag (announced by BA)
- [ ] Changes of Beitragsbemessungsgrenze for KV and RV
- [ ] Increase in Mindestunterhalt

**Comment by MImmesberger** [comment]
I am not aware of any more parameter changes missing. All the changes that are documented [here](https://www.bundesregierung.de/breg-de/service/gesetzesvorhaben/neuregelungen-januar-2024-2250794) and some other changes regarding social security are implemented.

The codecov complaint is a bug, probably because I removed a shebang in a module.

File: CHANGES.md [hunk 4]
```diff
@@ -6,6 +6,7 @@ releases are available on [Anaconda.org](https://anaconda.org/conda-forge/gettsi
 
 ## Unpublished
 
+- {gh}`680` 2024 Parameter Update ({ghuser}`Eric-Sommer`)
```
Comment (hmgaudecker) [inline CHANGES.md:9]:
```suggestion
- {gh}`680` 2024 Parameter Update ({ghuser}`Eric-Sommer`, {ghuser}`MImmesberger`)
```

File: src/_gettsim/parameters/kinderzuschl.yaml [hunk 5]
```diff
@@ -27,9 +27,12 @@ maximum:
     reference: Art. 6 G. v. 08.12.2022 BGBl. I S. 2230.
     note: Regelung für das Kalenderjahr 2023, abweichend von § 6a Absatz 2.
   2024-01-01:
-    scalar: null
-    reference: § 6a (2) BKGG
-    note: Ab 2024 wird der Betrag wieder aus anderen Parametern berechnet.
+    scalar: 292
+    reference: null
+    note: >-
+      Calculated form other parameters (as 2021/2022). Parameters not yet known.
```
Comment (hmgaudecker) [inline src/_gettsim/parameters/kinderzuschl.yaml:33]:
```suggestion
      No law, should be calculated from other parameters (as 2021/2022), but parameters are not known yet.
```

File: src/_gettsim/parameters/kinderzuschl.yaml
Comment (hmgaudecker) [inline src/_gettsim/parameters/kinderzuschl.yaml:1]: Please check the Kindersofortzuschlag. I am pretty sure it cannot be gone in 2023 and be back in 2024.

**Review by hmgaudecker — COMMENTED** [review]
Looks good! 

- I did not check the actual values. Please make sure someone double-checks, unless you have already done so
- One important check left (kindersofortzuschlag)
- For the future, we should have a clear protocol of how to do this, maybe a checklist with typical things to adjust. Would it make sense to immediately add tests then, too?

File: src/_gettsim/parameters/kinderzuschl.yaml
Comment (MImmesberger) [reply src/_gettsim/parameters/kinderzuschl.yaml:1]: Thanks, didn't know that this exists.
The Sofortzuschlag for Bürgergeld (in `arbeitsl_geld_2.yaml`) is correct. But indeed, the Sofortzuschlag for KiZ is included in the maxmimum value of the Kinderzuschlag from 2023 onwards.

**Comment by MImmesberger** [comment]
I added tests for

- max Kinderzuschlag
- Minijobgrenze
- Bürgergeld
- Soli
- Sozialversicherungsbeiträge
- Einkommensteuer

This should cover the most important changes. I noticed that also the mean Zusatzbeitrag of Krankenversicherung changed.

I left out taxable income (changed because of Freibeträge) because the BMF calculator doesn't have the option to calculate it for 2024 yet.

I have one problem with the ALG2 tests because GETTSIM and the [calculator that we used in the past](https://www.buerger-geld.org/rechner/) do not agree on the relevant income for ALG2 calculation. I prepared the test but let pytest skip it for now because I don't know whether the error is on their or on our end. However, the calculator and GETTSIM agree on ALG2 transfers for individuals with no income. Once the BMF calculator is available we can check that.

**Comment by MImmesberger** [comment]
And I just realized that my workaround to test `minijob_grenze` (adding a fake dependency as discussed in #645) doesn't work because that clashes with the current tutorial. I'd suggest we skip those tests for now (I checked that everything is correct) until #666 is fixed.

**Comment by hmgaudecker** [comment]
> > we should have a clear protocol of how to do this
> 
> I will create a discussion or issue for that. Should be really helpful in the future.
> 

I think the way to go would be to make a PR template, starting from the [to-do list above](https://github.com/iza-institute-of-labor-economics/gettsim/pull/680#issuecomment-1900712978).

**Review by hmgaudecker — APPROVED** [review]
Thanks a lot!

**Pull request merged**